boxoffice/views/admin_discount.py

```diff
@@ -0,0 +1,210 @@
+        if not discount_price_form.validate_on_submit():
+            return api_error(message=discount_price_form.errors, status_code=400)
+    discount_policy_form.populate_obj(discount_policy)
+    discount_policy.make_name()
+    db.session.add(discount_policy)
+    db.session.commit()
+
+    if discount_policy.is_price_based:
+        discount_price = Price(discount_policy=discount_policy)
+        discount_price_form.populate_obj(discount_price)
+        discount_price.make_name()
+        db.session.add(discount_price)
+        discount_policy.items.append(discount_price.item)
+        db.session.commit()
+    return api_success(result={'discount_policy': jsonify_discount_policy(discount_policy)},
+        doc="New discount policy created.", status_code=200)
+
+
+@app.route('/admin/o/<org>/discount_policy/<discount_policy_id>/edit', methods=['OPTIONS', 'POST'])
+@lastuser.requires_login
+@load_models(
+    (Organization, {'name': 'org'}, 'organization'),
+    (DiscountPolicy, {'id': 'discount_policy_id'}, 'discount_policy'),
+    permission='org_admin'
+    )
+@xhr_only
+def admin_edit_discount_policy(organization, discount_policy):
+    discount_policy_form = DiscountPolicyForm(request.form)
+    all_items = Item.query.join(ItemCollection).filter(
+        ItemCollection.organization == organization).options(db.load_only('id')).all()
+    discount_policy_form.items.query = all_items
+
+    if not discount_policy_form.validate_on_submit():
+        return api_error(message=discount_policy_form.errors, status_code=400)
+
+    if not discount_policy.is_price_based:
+        discount_policy.items.extend(discount_policy_form.items.data)
+        discount_policy_form.populate_obj(discount_policy)
+    else:
+        discount_price_form = DiscountPriceForm(request.form)
+        discount_price_form.item.query = all_items
+        if not discount_price_form.validate_on_submit():
+            return api_error(message=discount_price_form.errors, status_code=400)
+        discount_price = Price.query.filter_by(discount_policy=discount_policy).one_or_none()
+        if discount_price:
+            discount_price_form.populate_obj(discount_price)
+            discount_policy_form.populate_obj(discount_policy)
+            discount_policy.items.append(discount_price.item)
+    db.session.commit()
+    return api_success(result={'discount_policy': jsonify_discount_policy(discount_policy)}, doc="Discount policy updated.", status_code=200)
+
+
+@app.route('/admin/discount_policy/<discount_policy_id>/coupons/new', methods=['OPTIONS', 'POST'])
+@lastuser.requires_login
+@load_models(
+    (DiscountPolicy, {'id': 'discount_policy_id'}, 'discount_policy'),
+    permission='org_admin'
+    )
+@xhr_only
+def admin_create_coupon(discount_policy):
+    coupon_form = DiscountCouponForm()
+    coupons = []
+    if not coupon_form.validate_on_submit():
+        return api_error(message=coupon_form.errors, status_code=400)
```

Comment:
This is incorrect use of `message`. The message should always be a human readable message, with a separate `errors` parameter enumerating the fields and error messages.